src/ogr/feature.jl

```diff
@@ -402,13 +455,25 @@ const _FETCHFIELD = Dict{OGRFieldType,Function}(
     OFTInteger64List => asint64list,
 )
 
+"""
+    getfield(feature, i)
+
+When the field is unset, it will return `nothing`. When the field is set but
+null, it will return `missing`.
+
+### References
+* https://gdal.org/development/rfc/rfc53_ogr_notnull_default.html
+* https://gdal.org/development/rfc/rfc67_nullfieldvalues.html
+"""
 function getfield(feature::Feature, i::Integer)
-    return if isfieldset(feature, i)
-        _fieldtype = gettype(getfielddefn(feature, i))
+    return if isfieldsetandnotnull(feature, i)
+        _fieldtype = getfieldtype(getfielddefn(feature, i))
         _fetchfield = get(_FETCHFIELD, _fieldtype, getdefault)
         _fetchfield(feature, i)
+    elseif isfieldset(feature, i)
+        missing
     else
-        getdefault(feature, i)
+        nothing
     end
 end
```

Comment:
> Since we want to distinguish unset and null state, shouldn't we use isfieldset and isfieldnull instead ?

Fixed in https://github.com/yeesian/ArchGDAL.jl/pull/238/commits/323cbdb7719a1ad62affec6d0462fd0de7d18d7b.

> And since, we should not have any unknown fieldtype, shouldn't we raise an error in case of unknown type instead

Fixed in https://github.com/yeesian/ArchGDAL.jl/pull/238/commits/5c7b677e5350fe581a1e5ec9559d9ac91d302908.